### PR TITLE
Add now as optional property.

### PR DESCRIPTION
**Pull request by aljones15**
- Upgrade eslint.
  - Add jsdoc comments to library.

NOTE: this requires the next version of `http-signature-header`.

sister PR:
https://github.com/digitalbazaar/http-signature-zcap-invoke/pull/12

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -12,11 +12,40 @@ import {parseRequest, parseSignatureHeader} from 'http-signature-header';
 import {CapabilityInvocation} from 'ocapld';
 import {TextDecoder, TextEncoder, base64Decode} from './util.js';
 
+/**
+ * Verifies a zcap in the form of an http-signature header.
```
Comment (dlongley) [inline main.js]:
```suggestion
 * Verifies a zcap invocation in the form of an http-signature header.
```

File: main.js [hunk 2]
```diff
@@ -12,11 +12,40 @@ import {parseRequest, parseSignatureHeader} from 'http-signature-header';
...
+ *
+ * @param {object} options - Options to use.
+ * @param {URL} options.url - The url of the request.
```
Comment (dlongley) [inline main.js]: Isn't this a string?

```suggestion
 * @param {string} options.url - The url of the request.
```

File: main.js [hunk 3]
```diff
@@ -12,11 +12,40 @@ import {parseRequest, parseSignatureHeader} from 'http-signature-header';
...
+ * @param {string} options.method - The HTTP request method.
+ * @param {Array<string>} options.headers - The headers used to make the
+ *   signature.
```
Comment (dlongley) [inline main.js]: We don't yet know which headers were used to make the signature, right? These are just the headers we received in the request (is my understanding).

```suggestion
 * @param {Array<string>} options.headers - The headers from the request.
```

File: main.js [hunk 4]
```diff
@@ -12,11 +12,40 @@ import {parseRequest, parseSignatureHeader} from 'http-signature-header';
...
+ * @param {Function<Promise>} options.getInvokedCapability - An async
+ *   function that can dereference the capability.
```
Comment (dlongley) [inline main.js]:
```suggestion
 * @param {Function<Promise>} options.getInvokedCapability - An async
 *   function to call to dereference the invoked capability if it was passed
 *   by reference.
```

File: main.js [hunk 5]
```diff
@@ -12,11 +12,40 @@ import {parseRequest, parseSignatureHeader} from 'http-signature-header';
...
+ * @param {Function} options.documentLoader - A jsonld documentloader.
+ * @param {string} options.expectedHost - The expected host of the request.
+ * @param {string} options.expectedTarget - The expected target of the zcap.
+ * @param {string} options.expectedRootCapability - The expected root capability
+ *   of the zcap.
+ * @param {string} options.expectedAction - The expected allowed action of the
+ *  zcap.
+ * @param {Function} options.inspectCapabilityChain - A function that can
+ *   validate a capability chain.
```
Comment (dlongley) [inline main.js]:
```suggestion
 *   inspect a capability chain.
```

**Review by dlongley — APPROVED** [review]
Thanks! Left some documentation suggestions to apply.

**Review by OR13 — APPROVED** [review]

**Comment by mattcollier** [comment]
@aljones15 it looks like @dlongley made several suggestions that need to be addressed here.

**Comment by OR13** [comment]
plz merge :)

**Comment by aljones15** [comment]
@OR13 once the next version of http signature header is released this is good to go. Sorry about the wait.

**Comment by aljones15** [comment]
@OR13 Mr. Steele. Where are your test vectors? This branch is ready for them. If we could get them and test them using these branches first that might be better as we can fix problems with our implementation before making a major release. NOTE: this PR is pinned to 2 branches one for `http-signature-header` and the second for `http-signature-zcap-invoke`. Do not release until all branches are removed from the package.json with releases of those respective libraries.

**Comment by OR13** [comment]
@aljones15 I was going to use your libraries to generate stable test vectors and have been waiting for the ability to do that, can you give me the specific package.json entries I can use to produce them, and I will generate some.

**Comment by aljones15** [comment]
> @aljones15 I was going to use your libraries to generate stable test vectors and have been waiting for the ability to do that, can you give me the specific package.json entries I can use to produce them, and I will generate some.

the `package.json` in this branch should be enough.

I believe what you need it is:

```js
npm i -S digitalbazaar/http-signature-zcap-verify#add-now-to-parse-request
npm i -S digitalbazaar/http-signature-zcap-invoke#set-create-expires-11
npm i -S digitalbazaar/http-signature-header
```
that should allow you to pass in `now`, `created` and `expires` as unix timestamps. Let me know if you need anything else.

**Comment by OR13** [comment]
@aljones15 thanks I am not sure when I will get to it, but I will cross link so it does not get forgotten.

**Comment by aljones15** [comment]
> @aljones15 thanks I am not sure when I will get to it, but I will cross link so it does not get forgotten.

if that is the case we might want to release before you get around to it. I was hoping for the additional tests before we released, but if there will be a wait might be better if we verify on our end. Is there anywhere I can see what you are going to test?

**Comment by OR13** [comment]
@aljones15 release early and often... If I were you I would release asap :)

**Comment by mattcollier** [comment]
@dlongley @aljones15 Related to: https://github.com/digitalbazaar/http-signature-header/issues/41

There's no changelog entry included in this PR, I'm trying to determine how this should be released.  Please provide additional details and I can update the changelog and release as necessary.

**Review by mattcollier — APPROVED** [review]
tested successfully with the bedrock-profile test suite.

**Pull request merged**